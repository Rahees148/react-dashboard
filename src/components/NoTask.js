
import styled from 'styled-components';

//Context
import {useContext} from 'react';
import {TaskContext} from '../context/TaskContext';


const Wrapper = styled.section`
  width: 100vw;
  height: 100vh;
  padding-top:90px;
  @media (min-width: 768px) {
    padding-top:0px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;
const NoTaskContainer = styled.div`
  width:100%;
  padding:24px;
  text-align:center;
  background-color:#ffffff;
  box-shadow: 0px 3px 6px #00000029;
  
  @media (min-width: 768px) {
    border-radius: 12px;
    max-width:300px;
  }
`;
const Title = styled.h1`
  color: #537178;
  font-size:20px;
  line-height:28px;
  font-weight:500;
  color: #537178;
  margin:0px 0 24px;
`;

const Button = styled.button`
  background-color: #5285EC;
  color:#ffffff;
  border-radius:8px;
  padding:11px 22px;
  text-align:center;
  outline:none;
  cursor: pointer;
`;

function NoTask(){
    const context = useContext(TaskContext);
    return (
       
        <Wrapper>
            <NoTaskContainer>
                <Title>You have no task.</Title>
                <Button onClick={context.toggleNewTask}>+ New Task</Button>
            </NoTaskContainer>
        </Wrapper>
        
    )
}

export default NoTask;