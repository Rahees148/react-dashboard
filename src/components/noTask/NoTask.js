import { Style } from "./NoTask.style";

//Context
import { useContext } from "react";
import { TaskContext } from "../../context/TaskContext";
//Atoms
import Button from "../atoms/button/Button";
import Title from '../atoms/title/Title';

function NoTask() {
  const context = useContext(TaskContext);
  return (
    <Style.Wrapper>
      <Style.NoTaskContainer>
        <Title>You have no task.</Title>
        <Button onClick={context.toggleNewTask}>+ New Task</Button>
      </Style.NoTaskContainer>
    </Style.Wrapper>
  );
}

export default NoTask;
