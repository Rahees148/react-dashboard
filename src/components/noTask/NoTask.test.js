
import renderer from 'react-test-renderer';
import NoTask from './NoTask';


it('renders correctly', () => {
  const tree = renderer
    .create(<NoTask />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});