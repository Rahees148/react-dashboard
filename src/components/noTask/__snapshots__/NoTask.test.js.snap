// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-bdfBwQ bUcYFa"
>
  <div
    className="sc-gsTCUz hwMqDN"
  >
    <h1
      className="sc-dlfnbm dsDWFB"
    >
      You have no task.
    </h1>
    <button
      className="sc-hKgILt ireMci"
      onClick={[Function]}
    >
      + New Task
    </button>
  </div>
</section>
`;
